replica/table.cc

```diff
@@ -754,6 +774,7 @@ table::start() {
 
 future<>
 table::stop() {
+    _sstables_changed.signal();
```

Comment:
I'd suggest signaling _sstables_changed after '_sstables = make_compound_sstable_set();', so the wait loop will be able to bail out. could happen that we signal it here, and when loop wakes up, sstable set still doesn't satisfy the condition, so wait happens, table is destroyed, and we end up with use-after-free.